doc/chromebooks/luks-installation.md

```diff
@@ -0,0 +1,433 @@
+```
+blkid
+```
+output:
+```
+/dev/mmcblk0p3: LABEL="bootemmc" UUID="ae2dfad1-f078-40f3-9542-defb28394228" BLOCK_SIZE="4096" TYPE="ext4" PARTUUID="a4621290-1765-406b-bf80-92d91a313815"
+/dev/mmcblk0p4: UUID="5e9ef95e-3b51-4525-898f-419a0b0a6e4f" TYPE="crypto_LUKS" PARTUUID="086cb1ce-1ddb-4304-811e-5e1752836da5"
+/dev/mapper/encrypted: LABEL="rootemmc" UUID="f9c4038c-3625-45f5-9510-94ddc1bb7b23" UUID_SUB="ab69e1dc-5a73-4f0b-8dee-d5b87586d9b5" BLOCK_SIZE="4096" TYPE="btrfs"
+/dev/mmcblk0p1: PARTLABEL="KernelA" PARTUUID="88b9134a-9886-304f-a735-a97301c40284"
+/dev/mmcblk0p2: PARTLABEL="KernelB" PARTUUID="bad61d81-12d2-4c47-81d6-89e2f0f4157e"
+```
+find the partition UUID and copy it or note down
+
+_Info. in our case we are looking for UUID of an encrypted (not mapped one) partition with TYPE "crypto_LUKS" which in this example is "5e9ef95e-3b51-4525-898f-419a0b0a6e4f"_
+
+_Warning. make sure you aren't looking at PARTUUID_
+
+now open the crypttab file
+```
+nano /mnt/etc/crypttab
+```
+and put in this
+```
+# <target name> <source device>         <key file>      <options>
+encrypted UUID=5e9ef95e-3b51-4525-898f-419a0b0a6e4f none luks,discard
+```
+_Note. remember to adjust the UUID to yours_
+
+- Enable xz compression for initrd (in theory optional but recommended)
+
+_Note. saves space in /boot and makes initrd smaller for fit image in case that is needed - memlimit might need adjustment in case update-initramfs gives an error_
+
+open the /etc/initramfs-tools/conf.d/compress file
+```
+nano /mnt/etc/initramfs-tools/conf.d/compress
+```
+and put this inside
+```
+COMPRESS=xz
+XZ_OPT='-9 --check=crc32 –memlimit-compress=25%'
+```
+
+9. final adjustments
+
+start by entering the chroot
+```
+chroot /mnt mount -t proc proc /proc 
+chroot /mnt mount -t sysfs sysfs /sys 
+mount --bind /dev /mnt/dev 
+mount --bind /run /mnt/run
+chroot /mnt
+```
+_Note. this will make it so, the target disk will be treated like a root directory in that terminal as long as you don't type ```exit```_
+
+_Note. after entering the command, if you want to be sure you are still in chroot, you can type ```export PS1="(chroot) ${PS1}"``` (it doesn't check for being inside chroot but gives visual indicator untill u exit it)_
+
+<details>
+<summary>if you messed up and want to enter chroot again after reboot
+</summary>
+
+you need to export the drive and part from part one again (look up)
+and mount the partitions again
+```
+#root
+cryptsetup open --type luks /dev/${part}4 encrypted
+mount -o ssd,compress-force=zstd,noatime,nodiratime /dev/mapper/encrypted /mnt
+#boot
+mount /dev/${part}3 /mnt/boot
+```
+and then run commands to enter chroot again
+</details>
+
+now in the /boot directory create as "simple script" :3
+```
+nano /boot/kernel-and-initrd.sh
+```
+<details>
+<summary>here is the script</summary>
+
+```
+#!/bin/bash
+
+if [ "$(whoami)" != "root" ]; then
+  echo "This script must be ran as root."
+  exit 1
+fi
+
+cd /boot || exit
+
+# choosing kelner version
```

Comment:
can we please change "kelner" to "kernel" here everywhere and in the provided script? - i know that you usually use this name, but naming it "kernel" as usual it might be more clear to others what is ment

Reply:
yeah it wasn't on purpose
it's just the fact i have dyslexia
and i keep swapping l with r since these sound similar in my language and 'm never able to figure out which is first unless i look it up

will fix it